.github/workflows/checks.yml

```diff
@@ -72,67 +103,43 @@ jobs:
       - name: Setup environment
         uses: ./.github/actions/setup-env
 
-      - name: Read Changesets Config
-        id: changeset-config
-        run: |
-          BASE_BRANCH=$(jq -r '.baseBranch' .changeset/config.json)
-          echo "base_branch=$BASE_BRANCH" >> $GITHUB_OUTPUT
-
-      - name: Determine Release Details
-        id: release-details
-        # runs only when the .changeset.baseBranch is the target branch
-        if: steps.changeset-config.outputs.base_branch == '${{ github.ref_name }}'
-        run: |
-          OUTPUT=$(mktemp ${{runner.temp}}/changeset-output.XXXXXX.json)
-
-          if [ ! -f "$OUTPUT" ]; then
-            echo "Failed to create temp file at: $OUTPUT"
-            exit 1
-          fi
-          echo "Temporary file created at: $OUTPUT"
-
-          # the `changeset status` command uses path.join so the path should be relative
-          yarn changeset status --output $(realpath --relative-to . $OUTPUT)
-
-          if [ ! -s "$OUTPUT" ]; then
-            echo "Error: Temporary file is empty after running 'yarn changeset status'"
-            exit 1
-          fi
```

Comment:
that part has been extracted to `.github/actions/changeset-status/action.yml` for convenient use in both `checks` and `build` jobs